font-types/src/raw.rs

```diff
@@ -159,6 +159,10 @@ impl<T: Scalar> FixedSize for BigEndian<T> {
     const RAW_BYTE_LEN: usize = T::RAW_BYTE_LEN;
 }
 
+impl FixedSize for () {
+    const RAW_BYTE_LEN: usize = 0;
```

Comment:
I'm worried about this; there are various places where we do `bytes.len() / T::RAW_BYTE_LEN`, and we really don't expect this to be zero, even if we're doing checked ops.

I'm sure there will be an interesting explanation coming up, but I want to note my concern up front.


_much later:_

okay, coming back to this cliffhanger after getting through everything... I have a general sense of what you're trying to do, and although I don't have a good enough sense of what the allowed values of a `StateEntry` are, _assuming_ we go with basically the approach you've taken here, I'd like to do it with a custom sealed type instead of with `()`.

Basically: this is okay as a hack as long as it is an invariant that the zero-sized `FixedSize` type is only ever used as a component of some other type, and has no public constructors.

as a first sketch, I might try:

```rust
/// An empty type for use exclusively in `StateEntry`.
// NOTE: the inner field is private, and this type cannot be constructed outside of this module
pub struct NoPayload(());
```

Reply:
I was wondering if an impl of FixedSize for a zero-sized type might bite us. I’ll try the private type as you suggest.

The payload is going to be one of _nothing_, a u16 or a pair of u16s (I think there’s an example of this one in the tests). 

Looking forward, state tables will always be embedded directly in other tables so we won’t be able to use codegen anyway. I might just eschew the FontRead impl and add a private method so we don’t have to worry about the implications of these in the public API.